### PR TITLE
Create search component using Algolia DocSearch V3

### PR DESCRIPTION
**Pull request by codingone21**
Algolia Docsearch V3 has most setting auto-configured and pre-loaded. How to custom style their search is not documented yet, ([link](https://docsearch.algolia.com/docs/styling)) so I created a style.css specifically for DocSearch. I copied the style.css which is currently generated in their @docsearch/css@3 package, and it seemed to have worked :) We can update the style configuration here.

This PR is just for the design component Algolia Docsearch keys for Flow Doc Search project are hardcoded in the code, and it will be configured in the env in the future.

Screen Shot 2022-05-18 at 7 41 16 PM

### CODE REVIEW

File: libs/design-system/src/lib/Components/Search/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export type SearchProps = {
+  appId: string;
+  apiKey: string;
+  indexName: string;
+};
+
+export function Search({ appId, apiKey, indexName }: SearchProps) {
+  return (
+    <DocSearch
+      appId="QE0LM9XKDG"
```
[the remark this answers is not in the document]
Comment (10thfloor) [reply libs/design-system/src/lib/Components/Search/index.tsx:13]: You can create a `.env` file in the `apps/flow-docs` project. 
Once you have that take a look here for how to use these in client code via a Remix: https://remix.run/docs/en/v1/guides/envvars#browser-environment-variables

[the remark this answers is not in the document]
Comment (codingone21) [reply libs/design-system/src/lib/Components/Search/index.tsx]: I ran into a bunch of dependency issues trying to work with dotenv :( I'm gonna continue to debug, but I opened the PR for review just for the design part! This PR has the testing creds so no secrets are revealed.

**Review by alse — APPROVED** [review]
Looks good! If we want to use the tailwind theme/colors we'll have to import the style overrides in tailwind.css (main.css in the v2 app)

**Comment by 10thfloor** [comment]
Closing because code was migrated.

**Pull request closed without merging**